tests/integration/test_charm.py

```diff
@@ -321,12 +321,40 @@ def unit_uri(ip_address: str) -> str:
     return "mongodb://{}:{}/".format(ip_address, PORT)
 
 
+def fetch_primary(replica_set_hosts: List[str]) -> str:
+    """Returns IP address of current replica set primary."""
+    # connect to MongoDB client
+    if len(replica_set_hosts) == 1:
+        replica_set_uri = replica_set_hosts[0] + ":" + str(PORT)
+        client = MongoClient(replica_set_uri, replicaset="rs0")
+    else:
+        client = MongoClient(generate_replica_set_uri(replica_set_hosts))
+
+    # grab the replica set status
+    try:
+        status = client.admin.command("replSetGetStatus")
+    except (ConnectionFailure, ConfigurationError, OperationFailure):
+        return None
+    finally:
+        client.close()
+
+    primary = None
+    # loop through all members in the replica set
+    for member in status["members"]:
+        # check replica's current state
+        if member["stateStr"] == "PRIMARY":
+            # get member ip without ":PORT"
+            primary = member["name"].split(":")[0]
+
+    return str(primary)
```

Comment:
I would move these helper functions to a `helpers.py` file within `tests/integration` and import them.

That way we can keep all methods called within `test_charm.py` standard `test_XYZ`